sway-core/src/ir_generation/function.rs

```diff
@@ -482,6 +482,11 @@ impl FnCompiler {
                         .add_metadatum(context, span_md_idx))
                 }
             }
+            Intrinsic::AddrOf => {
+                let exp = arguments[0].clone();
+                let value = self.compile_expression(context, md_mgr, exp)?;
+                Ok(self.current_block.ins(context).addr_of(value))
```

Comment:
```suggestion
                Ok(self.current_block.ins(context).addr_of(value).add_metadatum(context, span_md_idx))
```
Need to add the span when creating instructions.